modules/openapi-generator/src/test/resources/3_0/issue_1517.yaml

```diff
@@ -50,6 +50,17 @@ paths:
           schema:
             type: string
             pattern: '/^[\pattern\d{3}$/i'
+        - name: alternation_with_forward_slash
+          in: header
+          schema:
+            type: string
+            pattern: '/ax$|/bx$'
+        - name: patten_starts_ends_with_slash
+          in: header
+          schema:
+            type: string
+            pattern: '/root/'
+            description: 'Should match only /root/ but not root'
```

Comment:
I added new test cases here rather than creating a separate file, but please let me know if there's a preference for keeping this separate